include/ginkgo/core/base/exception_helpers.hpp

```diff
@@ -147,6 +148,22 @@ inline dim<2> get_size(const T& op)
 inline dim<2> get_size(const dim<2>& size) { return size; }
 
 
+template <typename T>
+inline batch_dim<2> get_batch_size(const T& op)
```

Comment:
maybe just split this into two, one for the multivector, and one for linops. That would be a bit safer.

Reply:
Can you elaborate on what you mean by safer ? It would give a compile time error if `get_size` is not available, so I dont think there are any issues there ?

Reply:
IMO, the compiler error will be easier to understand if there are only two overloads. Otherwise you might even get linker instead of compiler errors.